tests/unit/vitest-sails.config.mjs

```diff
@@ -0,0 +1,28 @@
+import { defineConfig } from 'vitest/config';
+import { dirname, resolve } from 'path';
+import { fileURLToPath } from 'url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(resolve(__filename,'../..'));
+
+export default defineConfig({
+  resolve: {
+    alias: {
+      alias: {
+        _: __dirname,
+        '@': resolve(__dirname, './src'),
+      },
+    },
+  },
+  test: {
+    // vitest includes tests from the project root
+    include: ['tests/unit/specs/sails/**/*'],
+    globals: true,
+    environment: 'node',
+    setupFiles: ['tests/unit/setup-sails.vitest.js'],
```

Comment:
New sails specific file, that ONLY handles sails setup and runs the sails specific files. I think there's a way to parallelize these properly, but the point of this PR is to upgrade Vitest. I think improving this should be a follow up.